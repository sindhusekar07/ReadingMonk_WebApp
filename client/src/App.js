
import './pageStyle.css';

import React from 'react';
import './App.css';
//import axios from 'axios'
import Register from './Scripts/Register';
import HomePage from './Scripts/Homepage'
import AddBook from './Scripts/AddBook'
import SearchBook from './Scripts/SearchBook'
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import getOTP from "./Scripts/GetOTP"
import changePassword from "./Scripts/ChangePassword"
import Dashboard from './Scripts/Dashboard'
import Profile from './Scripts/Profile'
import userprofile from './Scripts/userprofile'


function App(props){
  React.useEffect(()=>{
    tracker();
  },[])
  var TimeOutFun;

  const NotIdle = ()=>{
    clearTimeout(TimeOutFun)
    TimeOutFun = setTimeout(()=>{
      if(window.location.pathname !== "/"){
        console.log(window.location.pathname)
        alert("Session Expired.")
        window.location.href="http://localhost:80/"
      }

    },900000)
  }
  const tracker = ()=>{
    window.addEventListener('click',()=>{
      NotIdle();
    })
    window.addEventListener('mousemove',()=>{
      NotIdle();
    })
    window.addEventListener('scroll',()=>{
      NotIdle();
    })
    window.addEventListener('keydown',()=>{
      NotIdle();
    })
  }
  return(    
    <Router>
      <Switch>
      <Route path='/' exact component={HomePage} />
      <Route path='/register' exact component={Register} />
      <Route path='/home' exact component={Dashboard} />
      <Route path="/pass_req" exact component={getOTP}/>
      <Route path="/passwordGet" exact component={changePassword}/>
      <Route path='/add_a_book' exact component={AddBook} />
      <Route path='/search_a_book' exact component={SearchBook} />
      <Route path='/profile' exact component={Profile} />
      <Route path='/userprofile' exact component={userprofile} />
      </Switch>     
    </Router>
  )
}

export default App;