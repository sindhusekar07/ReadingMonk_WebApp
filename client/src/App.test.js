import { render, fireEvent, cleanup } from '@testing-library/react';
import App from './App';
import React from 'react'
import LandingPage from './Scripts/LandingPage.js'
import RegisterScreen from './Scripts/Register'

//Render Action Testing
it('rendering without crashing', () => {});

//Snapshot testing
afterEach(cleanup)

it('should take a snapshot', () => {
    const { asFragment } = render(<App />)
    
    expect(asFragment(<App />)).toMatchSnapshot()
   });
//Button Click Testing
it("captures clicks", done => {
    function handleClick() {
      done();
    }
    const { getByText } = render(
      <button onClick={handleClick}>Click Me</button>
    );
    const node = getByText("Click Me");
    fireEvent.click(node);
  });
  
  afterEach(cleanup);

  //Input Chnage Event Testing

  it("captures changes", done => {
    function handleChange(evt) {
      expect(evt.target.value).toEqual("whamo");
      done();
    }
    const { getByPlaceholderText } = render(
      <input onChange={handleChange} placeholder="Change Me" />
    );
    const node = getByPlaceholderText("Change Me");
    fireEvent.change(node, { target: { value: "whamo" } });
  });
   

