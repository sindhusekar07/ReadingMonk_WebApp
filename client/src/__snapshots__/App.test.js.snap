// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    style="background-image: url(image2.jpg); display: flex; width: 100%; height: 100vh; flex-direction: column; justify-content: center; align-items: center;"
  >
    <div
      style="width: 40%; height: 50%; border-radius: 6px; display: flex; align-items: center; flex-direction: column; justify-content: space-around; background-color: white;"
    >
      <h1
        style="font-family: serif;"
      >
        Welcome to Book Exchange App!
      </h1>
      <input
        placeholder="Emai Id"
        style="width: 60%; height: 30px; font-weight: bold;"
      />
      <input
        placeholder="Password"
        style="width: 60%; height: 30px; font-weight: bold;"
        type="password"
      />
      <div
        style="display: flex; flex-direction: row; justify-content: space-around; align-items: center; width: 60%;"
      >
        <button
          style="width: 100px; height: 30px; border-radius: 5px; font-weight: bold; background-color: rgb(37, 142, 166); font-family: serif;"
        >
          Sign In
        </button>
        <button
          style="width: 100px; height: 30px; border-radius: 5px; font-weight: bold; background-color: rgb(37, 142, 166); font-family: serif;"
        >
          Sign Up
        </button>
        <a
          href="/pass_req"
          style="color: black; cursor: pointer; font-weight: bold; font-family: serif;"
        >
          Forget Password?
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
